src/Ledger/TokenAlgebra/ValueRelation.lagda

```diff
@@ -0,0 +1,471 @@
+\subsection{Value Relation}
+\label{sec:value-relation}
+
+
+\begin{code}[hide]
+{-# OPTIONS --safe #-}
+
+open import Agda.Primitive  using () renaming (Set to Type)
+open import Ledger.Epoch    using (GlobalConstants)
+
+module Ledger.TokenAlgebra.ValueRelation
+  (gc        : GlobalConstants)
+  (PolicyID  : Type)
+  where
+
+open import Algebra                      using (CommutativeMonoid ; Commutative ; Op₂)
+open import Algebra.Morphism             using (IsMonoidMorphism ; IsSemigroupMorphism)
+open import Data.Nat                     using (ℕ ; _≤_ ; _>_)
+                                         renaming (zero to 0ℕ ; _+_ to _+ℕ_)
+open import Data.Nat.Properties          using ( +-identityˡ ; +-identityʳ ; +-comm ; +-assoc ; +-0-commutativeMonoid )
+open import Data.Product.Base            using (swap)
+open import Data.Sum.Base                using ([_,_]′)
+
+open import Ledger.Prelude               hiding (Coin ; Rel ; _>_)
+open import Ledger.TokenAlgebra gc PolicyID
+                                         using (AssetID ; AssetName ; Quantity ; TokenAlgebraRel)
+open import Relation.Binary              using (REL ; Rel ; _⇒_ ; IsEquivalence ; Decidable)
+                                         renaming (_⇔_ to _⇐⇒_ )
+open import Relation.Binary.Definitions  using (Decidable ; DecidableEquality)
+
+open import Relation.Binary.PropositionalEquality.Core using ( module ≡-Reasoning )
+
+
+private
+  variable
+    A B : Type
+    C : REL A B 0ℓ → Type 0ℓ
+
+open ≡-Reasoning
+
+\end{code}
+
+\subsection{The identity of the value monoid}
+
+We'll use \AgdaBound{ι} to denote the identity for the Value monoid.
+
+\begin{code}
+ι : REL AssetID Quantity 0ℓ
+ι _ q = q ≡ 0
+
+ι-left-unique : left-unique-rel ι
```

Comment:
It's probably a good idea to put all those into a hidden code block (\begin{code}[hide]) and in a `private` block. We won't be using those anywhere, and I don't think it's worth showing those proofs in the PDF.